src/components/bulk-email-tool/bulk-email-task-manager/BulkEmailContentHistory.jsx

```diff
@@ -1,41 +1,241 @@
-import React, { useState, useEffect } from 'react';
+import React, { useState } from 'react';
+import PropTypes from 'prop-types';
 import { useParams } from 'react-router-dom';
+import { injectIntl, intlShape } from '@edx/frontend-platform/i18n';
 
-import { FormattedMessage } from '@edx/frontend-platform/i18n';
+import {
+  Alert, Button, DataTable, Modal,
+} from '@edx/paragon';
+import messages from './messages';
 import { getSentEmailHistory } from './api';
 
-export default function BulkEmailContentHistory() {
+export function BulkEmailContentHistory({ intl }) {
   const { courseId } = useParams();
 
-  const [emailHistoryData, setEmailHistoryData] = useState(); // eslint-disable-line no-unused-vars
+  const [emailHistoryData, setEmailHistoryData] = useState();
+  const [errorRetrievingData, setErrorRetrievingData] = useState(false);
+  const [showHistoricalEmailContentTable, setShowHistoricalEmailContentTable] = useState(false);
+  const [isMessageModalOpen, setIsMessageModalOpen] = useState(false);
+  const [messageContent, setMessageContent] = useState();
 
-  useEffect(() => {
-    async function fetchSentEmailHistoryData() {
-      const data = await getSentEmailHistory(courseId);
+  /**
+   * Async function that makes a REST API call to retrieve historical email message data sent by the bulk course email
+   * tool from edx-platform.
+   */
+  async function fetchSentEmailHistoryData() {
+    let data = null;
+    try {
+      data = await getSentEmailHistory(courseId);
+    } catch (error) {
+      setErrorRetrievingData(true);
+    }
+
+    if (data) {
       const { emails } = data;
       setEmailHistoryData(emails);
+      setShowHistoricalEmailContentTable(true);
+    }
+  }
+
+  /**
+   * This function is responsible for setting the current `messageContent` state data. This will be the contents of a
+   * previously sent email message from the bulk course email tool. This also toggles a modal to be visible to display
+   * the message contents to the end user.
+   */
+  const onViewMessageClick = (tableData) => {
+    setMessageContent(tableData);
+    setIsMessageModalOpen(true);
+  };
+
+  /**
+   * Render function for the email content history table. If an error occurs while attempting to fetch data from
+   * edx-platform we will render this error instead of the table.
+   */
+  const renderError = () => (
```

Comment:
What is the best way to co-locate a reusable component like this? Do we have a pattern I should follow?

Reply:
since I think this is something that would only be used by the `bulk-email-task-manager`, just as a single component in that folder that can be pulled in by the `contentHistory` and `pendingTasks` to display errors/empty state.